scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_model.py

```diff
@@ -287,6 +287,9 @@ def create_fit_tables(self, peak_centre_params=None):
             # axis for labels in workspace
             axis = TextAxis.create(nruns)
             for iws, wsname in enumerate(self._loaded_workspaces.keys()):
+                wsname_bgsub = wsname + "_bgsub"
```

Comment:
I think we can avoid this if the key in `model._loaded_workspaces` is the name workspace that correpsonds to the row in the table (i.e. if background subtraction has been applied, it is the name of background subtracted workspace). I think the only place this would cause a problem is when the user tries to load in the same data twice - but I think this is easier to handle because we can change the check in `load_files` to be something like `if any([key.startswith('generated_workspace_name') for key in self._loaded_workspaces])`